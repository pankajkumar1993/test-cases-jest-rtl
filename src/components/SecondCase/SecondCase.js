import { useState } from "react";

export const mockData = [
    {
        "id": 1,
        "first_name": "Fletcher",
        "last_name": "McVanamy",
        "email": "mmcvanamy0@e-recht24.de",
        "age": 30
    }, {
        "id": 2,
        "first_name": "Clarice",
        "last_name": "Harrild",
        "email": "charrild1@dion.ne.jp",
        "age": 35
    },
]
const SecondCase = () => {
    const [clicked, setClicked] = useState(false);

    const handleClick = () => {
        setClicked(true);
    };
    return (
        <div>
            <h2> First test </h2>

            <button onClick={handleClick}>Click me</button>
            {clicked && <p>Button clicked!</p>}

            <ul>
                {mockData.map(item => (
                    <li key={item.id}>
                        {item.id}
                        {item.first_name},
                        {item.last_name},
                        {item.email}
                    </li>
                ))}
            </ul>
        </div>
    )
}



export default SecondCase;