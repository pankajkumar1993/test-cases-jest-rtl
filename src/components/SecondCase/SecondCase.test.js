import { fireEvent, render, screen } from '@testing-library/react'
import SecondCase, { mockData } from './SecondCase';
import userEvent from '@testing-library/user-event';

test("SecondCase renders successfully", () => {
    render(<SecondCase />);
    const element = screen.getByText(/first test/i);
    expect(element).toBeInTheDocument();
})

test("List renered", () => {
    render(<SecondCase />)
    const element = screen.getByText(/Clarice/i);
    expect(element).toBeInTheDocument()
})


test("Check if the element exists", () => {
    render(<SecondCase />);
    const element = screen.getByRole('heading', { name: /first test/i });
    expect(element).toBeInTheDocument();
});


test("Renders list items correctly", () => {
    render(<SecondCase />);
    const firstItem = screen.getByText(/fletcher/i);
    const secondItem = screen.getByText(/clarice/i);

    expect(firstItem).toBeInTheDocument();
    expect(secondItem).toBeInTheDocument();
});


test("Renders correct number of list items", () => {
    render(<SecondCase />);
    const listItems = screen.getAllByRole('listitem');
    expect(listItems).toHaveLength(mockData.length);
});


test("Matches snapshot", () => {
    const { asFragment } = render(<SecondCase />);
    expect(asFragment()).toMatchSnapshot();
});




test("Renders different elements based on conditions", () => {
    render(<SecondCase />);
    const conditionalElement = screen.queryByText(/non-existent text/i);
    expect(conditionalElement).toBeNull();
});


// Basic Click Event:
test("Button click changes state", () => {
    render(<SecondCase />);

    // Assuming there's a button that changes the text on click
    const button = screen.getByRole('button', { name: /click me/i });
    fireEvent.click(button);

    const updatedText = screen.getByText(/button clicked/i);
    expect(updatedText).toBeInTheDocument();
});


// Test click with userEvent:
// test("Simulate user click using userEvent", () => {
//     render(<SecondCase />);

//     const button = screen.getByRole('button', { name: /click me/i });
//     userEvent.click(button);

//     const result = screen.getByText(/clicked/i);
//     expect(result).toBeInTheDocument();
// });


// // Test Click on List Item:
// test("Clicking on list item shows details", () => {
//     render(<SecondCase />);

//     // Click on a specific list item
//     const listItem = screen.getByText(/clarice harrild/i);
//     fireEvent.click(listItem);

//     // Check if the expected action occurs (e.g., details appear)
//     const details = screen.getByText(/clarice's details/i);
//     expect(details).toBeInTheDocument();
// });


// // Testing Multiple Clicks:
// test("Multiple clicks update count", () => {
//     render(<SecondCase />);

//     const button = screen.getByRole('button', { name: /increment/i });
//     const count = screen.getByTestId('count');

//     // Initial value check
//     expect(count).toHaveTextContent('0');

//     // Simulate multiple clicks
//     fireEvent.click(button);
//     fireEvent.click(button);

//     expect(count).toHaveTextContent('2');
// });


// // Click Event with Callback:
// test("Button click calls callback function", () => {
//     const mockCallback = jest.fn();

//     render(<button onClick={mockCallback}>Click me</button>);

//     const button = screen.getByRole('button', { name: /click me/i });
//     fireEvent.click(button);

//     expect(mockCallback).toHaveBeenCalledTimes(1);
// });


// test("Button click updates text", () => {
//     render(<SecondCase />);

//     const button = screen.getByText(/click me/i);
//     fireEvent.click(button);

//     const updatedText = screen.getByText(/button clicked/i);
//     expect(updatedText).toBeInTheDocument();
// });