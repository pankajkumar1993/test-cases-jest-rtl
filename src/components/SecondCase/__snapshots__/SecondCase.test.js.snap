// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
       First test 
    </h2>
    <button>
      Click me
    </button>
    <ul>
      <li>
        1Fletcher,McVanamy,mmcvanamy0@e-recht24.de
      </li>
      <li>
        2Clarice,Harrild,charrild1@dion.ne.jp
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
